.github/actions/make/action.yml

```diff
@@ -0,0 +1,35 @@
+name: Fetch or make an artifact
+description: Fetch an artifact with a given key. If it doesn't exist, run the given command and upload it.
+inputs:
+  key:
+    description: The artifact key
+    required: true
+  make-rule:
+    description: The GNU make rule which produces the artifact
+    required: true
+  target-path:
+    description: The relative path of the final artifact(s)
+    required: false
+    default: 'target/release'
+
+runs:
+  using: composite
+  steps:
+    - name: Attempt to download the artifact
+      uses: actions/cache/restore@v4
+      with:
+        key: ${{ inputs.key }}
+        path: "${{ inputs.target-path }}"
+      id: artifact-download
+
+    - name: Run make ${{ inputs.make-rule }} if needed
+      if: steps.artifact-download.outputs.cache-hit != 'true'
+      run: make ${{ inputs.make-rule }}
```

Comment:
We assume release but are not making with `RELEASE=1`. How did this ever work?

But probably we should make in release mode anyway.